### PR TITLE
Update src/library/Box/styled

### PR DESCRIPTION
**Pull request by tx-steven**
Update Box to allow new props:
`borderRadius` - responsive
`color (backgroundColor)`
`position`
`scrollable (overflow: 'scroll')`

### CODE REVIEW

**Review by tx-trent — CHANGES_REQUESTED** [review]
Add the new props to the propTypes and types file

**Review by tx-trent — APPROVED** [review]

File: src/library/Box/propTypes.js [hunk 1]
```diff
@@ -8,7 +8,9 @@ export const spacingPropType = oneOfType([
 ]);
 
 export const boxPropTypes = {
+  borderRadius: oneOfType([string, arrayOf(string)]),
```
Comment (tx-nestor) [inline src/library/Box/propTypes.js]: I'm okay with breaking the alpha-convention here and simply listing all of our custom props at the top to more easily distinguish them.

File: src/library/Box/propTypes.js [hunk 2]
```diff
@@ -8,7 +8,9 @@ export const spacingPropType = oneOfType([
...
   breakpoints: arrayOf(oneOfType([number, string])),
+  color: string,
```
Comment (tx-nestor) [inline src/library/Box/propTypes.js]: "color" usually refers to the text color. Does Mineral have a prop elsewhere that is specifically for background color? I'm assuming either `backgroundColor` or `bgColor` or `bg`.

File: src/library/Box/propTypes.js [hunk 3]
```diff
@@ -29,5 +31,7 @@ export const boxPropTypes = {
   padddingStart: spacingPropType,
   paddingTop: spacingPropType,
   paddingVertical: spacingPropType,
+  position: string,
+  scrollable: bool,
```
Comment (tx-nestor) [inline src/library/Box/propTypes.js]: `scrollable` is a shorthand convenience. We need to also support passthrough of `overflow` as well.

File: src/library/Box/styled.js [hunk 4]
```diff
@@ -59,14 +59,15 @@ const getSpacingStyles = (
 export const BoxRoot = styled('div', {
   shouldForwardProp: (prop) =>
     ['height', 'width'].indexOf(prop) === -1 && isPropValid(prop)
-})(({ breakpoints, height, inline, theme, width, ...restProps }) => {
+})(({ borderRadius, breakpoints, color, height, inline, position, scrollable, theme, width, ...restProps }) => {
```
Comment (tx-nestor) [inline src/library/Box/styled.js]: List our props first to better distinguish them.

File: src/library/Box/styled.js [hunk 5]
```diff
@@ -59,14 +59,15 @@ const getSpacingStyles = (
...
   const rtl = theme.direction === 'rtl';
 
   const mapValueToProperty = (
     property: string,
     value: SpacingValue
   ): number | string => {
     const map = {
+      borderRadius: (borderRadius) => borderRadius,
```
Comment (tx-nestor) [inline src/library/Box/styled.js]: You shouldn't need parens: `borderRadius => borderRadius`

File: src/library/Box/styled.js [hunk 6]
```diff
@@ -84,11 +85,15 @@ export const BoxRoot = styled('div', {
   };
 
   return {
+    ...(color && { backgroundColor: color }),
```
Comment (tx-nestor) [inline src/library/Box/styled.js]: Seems like `backgroundColor` would make more sense as a prop

File: src/library/Box/styled.js [hunk 7]
```diff
@@ -84,11 +85,15 @@ export const BoxRoot = styled('div', {
...
+    ...(position && { position }),
+    ...(scrollable && { overflow: 'scroll' }),
```
Comment (tx-nestor) [inline src/library/Box/styled.js]: Support `overflow` passthrough as well.

File: src/library/Box/types.js [hunk 8]
```diff
@@ -7,7 +7,9 @@ type SpacingSize = $Keys<typeof SPACING_SIZE>;
 type InsetSpacingSize = $Keys<typeof INSET_SPACING_SIZE>;
 
 export type BoxProps = {
+  borderRadius?: string | Array<string>,
```
Comment (tx-nestor) [inline src/library/Box/types.js]: list ours up top.

Also, what is this file? Is this Flow?

**Review by tx-nestor — CHANGES_REQUESTED** [review]

**Comment by tx-nestor** [comment]
@tx-steven Also, remember to rebase the branch to stem from v0.58.0

File: src/library/Box/propTypes.js (hunk 2 above)
Comment (tx-steven) [reply src/library/Box/propTypes.js]: I haven't seen them support color as a prop in any components. They seem to want you to rely on the theme for colors. I can change to backgroundColor to be more explicit about what we are changing.

File: src/library/Box/types.js (hunk 8 above)
Comment (tx-steven) [reply src/library/Box/types.js]: Yeah, this is all of their flow types.

**Pull request closed without merging**